src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,173 @@
+    fb->setName(QObject::tr("Feedback"));
+    fb->setDescription("Controls how much of the output signal is looped");
+    fb->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    fb->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    fb->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    fb->setDefault(0.0);
+    fb->setMinimum(-1.0);
+    fb->setMaximum(1.0);
+
+    EffectManifestParameter* range = manifest.addParameter();
+    range->setId("range");
+    range->setName(QObject::tr("Range"));
+    range->setDescription("Controls the frequency range across which the notches sweep.");
+    range->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    range->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    range->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    range->setDefault(0.5);
+    range->setMinimum(0.0);
+    range->setMaximum(1.0);
+
+    EffectManifestParameter* stereo = manifest.addParameter();
+    stereo->setId("stereo");
+    stereo->setName(QObject::tr("Stereo"));
+    stereo->setDescription(QObject::tr("Enables/disables stereo"));
+    stereo->setControlHint(EffectManifestParameter::CONTROL_TOGGLE_STEPPING);
+    stereo->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    stereo->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    stereo->setDefault(0);
+    stereo->setMinimum(0);
+    stereo->setMaximum(1);
+    return manifest;
+}
+
+PhaserEffect::PhaserEffect(EngineEffect* pEffect,
+                           const EffectManifest& manifest) 
+        : m_pStagesParameter(pEffect->getParameterById("stages")),
+          m_pLFOFrequencyParameter(pEffect->getParameterById("lfo_frequency")),
+          m_pDepthParameter(pEffect->getParameterById("depth")),
+          m_pFeedbackParameter(pEffect->getParameterById("feedback")),
+          m_pRangeParameter(pEffect->getParameterById("range")),
+          m_pStereoParameter(pEffect->getParameterById("stereo")) {
+    Q_UNUSED(manifest);
+}
+
+PhaserEffect::~PhaserEffect() {
+    //qDebug() << debugString() << "destroyed";
+}
+
+void PhaserEffect::processChannel(const ChannelHandle& handle,
+                                  PhaserGroupState* pState,
+                                  const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                  const unsigned int numSamples,
+                                  const unsigned int sampleRate,
+                                  const EffectProcessor::EnableState enableState,
+                                  const GroupFeatureState& groupFeatures) {
+
+    Q_UNUSED(handle);
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pLFOFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedbackParameter->value();
+    CSAMPLE range = m_pRangeParameter->value();
+    int stages = 2 * m_pStagesParameter->value();
```

Comment:
You can use here a switch control like Nicu has proposed to get around the scaling issue. 
Otherwise you need a region that snaps to a desired number of stages.